kv/main.go

```diff
@@ -328,15 +329,64 @@ func setupCockroach(parsedURL *url.URL) (database, error) {
 	if *splits > 0 {
 		r := rand.New(rand.NewSource(int64(time.Now().UnixNano())))
 		g := newHashGenerator(&sequence{val: *writeSeq, seed: *seqSeed})
+
+		splitPoints := make([]int64, *splits)
 		for i := 0; i < *splits; i++ {
-			if _, err := db.Exec(`ALTER TABLE test.kv SPLIT AT VALUES ($1)`, g.hash(r.Int63())); err != nil {
-				return nil, err
-			}
+			splitPoints[i] = g.hash(r.Int63())
 		}
+		sort.Slice(splitPoints, func(i, j int) bool {
+			return splitPoints[i] < splitPoints[j]
+		})
 
-		if _, err := db.Exec(`ALTER TABLE test.kv SCATTER`); err != nil {
-			return nil, err
+		type pair struct {
+			lo, hi int
+		}
+		splitCh := make(chan pair, *concurrency)
+		splitCh <- pair{0, len(splitPoints)}
+		doneCh := make(chan struct{})
+
+		var wg sync.WaitGroup
+		for i := 0; i < *concurrency; i++ {
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				for {
+					p, ok := <-splitCh
+					if !ok {
+						break
+					}
+					m := (p.lo + p.hi) / 2
+					split := splitPoints[m]
+					if _, err := db.Exec(`ALTER TABLE test.kv SPLIT AT VALUES ($1)`, split); err != nil {
+						log.Fatal(err)
+					}
+					if _, err := db.Exec(fmt.Sprintf(`ALTER TABLE test.kv SCATTER FROM (%d) TO (%d)`,
+						splitPoints[p.lo], split+1)); err != nil {
```

Comment:
Minor, but a comment here would make this clearer. I assume the `+1` on `split` is to make it hit both sides of the split, but it took some thinking to understand the `+1`.